src/main/kotlin/com/autonomousapps/subplugin/ProjectPlugin.kt

```diff
@@ -592,38 +592,30 @@ internal class ProjectPlugin(private val project: Project) {
       return
     }
 
+    if (pluginManager.hasPlugin(SPRING_BOOT_PLUGIN)) {
+      logger.warn(
+        "(dependency analysis) You have both java-library and org.springframework.boot applied. You probably " +
+                "want java, not java-library."
+      )
+    }
```

Comment:
Moved up from below. Missed this in #852 and the warning would currently be emitted several times (once per source set) which makes no sense. There is no test for this warning. Do you want to keep it or should we remove it?

Turns out that this is unrelated to the fix. The use case most (all?) folks have is combining these three plugins: `java`, `springframework.boot`, `java-test-fixtures`. If you do that, you won't get the warning.

Reply:
If you think the warning is correct, let's leave it. Probably worth adding a test (which I can do later).